### PR TITLE
Hide zero inventory

### LINKED ISSUE #282: Add toggle key for hiding 0 in inventory
From #49 - let's add a toggle key that would hide/show entities with count 0.
This could shorten the length of the inventory in some situations.

As an alternative we could:
- add a separate list of "Known entities" below installed devices
- abandon this idea and revive #126 instead

### PR DESCRIPTION
**Pull request by noahyor**
This should resolve #282 .

### CODE REVIEW

File: src/Swarm/Game/Step.hs
Comment (xsebek) [inline src/Swarm/Game/Step.hs:559]: Your main branch seems to be behind? Please sync your main with byorgey/main. :slightly_frowning_face: 

I usually branch from byorgey/main itself, but you should have a button in your repo that says **"Fetch upstrem"**.

That might do the trick, if not, rebase the last commit on your main :slightly_smiling_face:

File: src/Swarm/TUI/Model.hs
Comment (xsebek) [inline src/Swarm/TUI/Model.hs:389]: Missing comment :wink:

File: src/Swarm/TUI/Controller.hs
Comment (xsebek) [inline src/Swarm/TUI/Controller.hs:306]: I wonder, would it not be enough to check something like `showZero /= listShowsZero`?
I am basing that on `listRobotHash /= focusedRobotHash`, but I have not looked into the code. :slightly_smiling_face: 

If there is some good reason, you can mention it in the `uiInventoryShouldUpdate` docstring :wink:

**Review by xsebek — CHANGES_REQUESTED** [review]
Please merge the #302 - we use CI to avoid different code styles (which would produce diffs).

Also, the description of the PR will be used as a squashed commit by Mergify at the end.
But for this small change it is probably fine :slightly_smiling_face:

File: src/Swarm/TUI/Model.hs
Comment (noahyor) [reply src/Swarm/TUI/Model.hs:389]: What do you think I should change it to?

File: src/Swarm/Game/Step.hs
Comment (xsebek) [reply src/Swarm/Game/Step.hs:559]: Yes, but you did not update *your* main branch and then you branched from it :man_shrugging: 

It probably would not cause any issue when merging, but I would prefer keeping PRs clean of such noise. :slightly_smiling_face:

File: src/Swarm/TUI/Model.hs
Comment (xsebek) [reply src/Swarm/TUI/Model.hs:389]: Probably just mention what toggles it (and that it is used when viewing same robots inventory?), also see my next comment.

File: src/Swarm/Game/Step.hs
Comment (xsebek) [reply src/Swarm/Game/Step.hs:559]: See below, the blue branch is this one, white is your main and orange is byorgey/main:

Screenshot from 2022-02-01 20-47-34

What I would like is, if the white was same as blue (that is your fech upstream into your main) and this branch would go from that updated main instead of the branch you used in #298 (I guess use git rebase).

Sorry to pester you about this, if you don't know how to use rebase, don't worry about it and the merge will hopefully work.
But then please switch back to branching from main. :smile:

File: src/Swarm/TUI/Controller.hs
Comment (byorgey) [reply src/Swarm/TUI/Controller.hs:306]: There isn't anywhere that we currently remember something like `listShowsZero`.  I suppose we could add that as another field of the `uiInventory`, but the `uiInventory` is already complicated enough.  I can imagine that in the future we might add other features that would require the inventory list to be updated, in which case we can reuse `uiInventoryShouldUpdate`, whereas something like `showZero /= listShowsZero` is specific to just this toggle.

File: src/Swarm/TUI/Model.hs
Comment (byorgey) [inline src/Swarm/TUI/Model.hs:385]:
```suggestion
-- | A toggle to show or hide inventory items with count 0 by pressing `0`
```

**Review by byorgey — APPROVED** [review]

**Comment by byorgey** [comment]
@xsebek Do you have time to take a quick look at this again and make sure the requested changes look good?  We can't merge it until you approve. =)

**Comment by byorgey** [comment]
I'm just going to go ahead and merge this manually, it looks to me like all the requested changes have been addressed.

**Pull request merged**
